tests/validate-urls.rb

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'English'
+require 'json'
+@status = 0
+
+# Fetch created/modified files in entries/**
+diff = `git diff --name-only --diff-filter=AM origin/master...HEAD entries/`.split("\n")
+
+def curl(url)
+  puts `curl --fail -sSI #{url}`
+  # rubocop:disable Style/GuardClause
+  unless $CHILD_STATUS.success?
+    # Break build if above cURL exited with non-zero value
+    puts "::error file=#{@path}:: Unable to reach #{url}"
+    @status = 1
+  end
+  # rubocop:enable Style/GuardClause
+end
+
+diff&.each do |path|
+  # Make path global for curl()
+  @path = path
+  entry = JSON.parse(File.read(@path)).values[0]
+
+  # Process the url,domain & additional-domains
+  curl((entry.key?('url') ? entry['url'] : "https://#{entry['domain']}/"))
+  entry['additional-domains']&.each { |domain| curl("https://#{domain}/") }
+
+  # Process documentation and recovery URLs
+  curl(entry['documentation']) if entry.key? !entry['documentation'].start_with?('/notes/')
```

Comment:
```suggestion
  if entry.key? 'documentation'
    unless entry['documentation'].start_with?('/notes/')
      curl(entry['documentation'])
    end
  end
```

Reply:
@ApCoder123 what's the reasoning? From what I can tell the suggested change does the same as the existing line but with more lines.

Reply:
When testing this locally, the documentation link wasn't actually being tested because the current line is missing the `'documentation'` parameter with the `entry.key?` method call. 

When adding this in with the line `if entry.key? 'documentation' && !entry['documentation'].start_with?('/notes/')`, the link still wasn't being tested. After changing to the current code, the documentation link was being tested - I did try shorter versions, but rubocop wasn't satisfied.

Reply:
@ApCoder123 it should work now.
I also set an old user-agent because some sites would 403 on the curl user-agent.